src/PageContent/RequestPickup/RecurringPickupRequest.js

```diff
@@ -0,0 +1,284 @@
+            formIsValid = false;
+            errors['time'] = 'Invalid time range';
+        }    
+
+        //Recur times
+        if(fields['recurTimes'] <= 0 && fields['endDate'] !== '') {
+            formIsValid = false;
+            errors['recurTimes'] = 'Pickup must recur at least once';
+        }
+
+        if((fields['recurTimes'] === '' && fields['endDate'] === '') 
+            || (!fields['recurTimes'] && !fields['endDate'])){
+            formIsValid = false;
+            errors['recurTimes'] = 'Must enter value for recurring times or an end date';
+        }
+
+        this.setState({errors: errors});
+        return formIsValid;
+    }
+
+    handleChange(field, e){         
+        let fields = this.state.fields;
+        fields[field] = e.target.value;        
+        this.setState({fields});
+    }
+
+    toggleModal(){
+        this.setState({
+            isOpen: !this.state.isOpen
+        });
+    }
+
+    // write to firebase
+    // var newRequest = firebase.database().ref().child("delivery_requests").push();
+    // newRequest.set(deliveryRequest);
+    
+    submitRequest(){
+        
+    }
+
+    createRequest(event) {
+        
+        if(!this.handleValidation()){ 
+            event.preventDefault();
+            alert('Form has errors');
+        }else{
+            event.preventDefault();
+            var raUid = event.target.receiveingAgency.value;
+            if (!raUid) raUid = null;
+            
+            var dgUid = event.target.delivererGroup.value;
+            if (!dgUid) dgUid = null;
+
+            var durationValue = null;
+            if (event.target.durationType.value === RequestDurationType.DATE) {
+                durationValue = event.target.endDate.value;
+            } else {
+                durationValue = event.target.numRecurrences.value;
+            }
+
+            // create DeliveryRequest object
+            var deliveryRequest = {
+                startDate: event.target.startDate.value,
+                duration:{
+                    type: event.target.durationType.value,
+                    value: durationValue
+                },
+                repeats: event.target.repeats.value,
+                startTime: event.target.startTime.value,
+                endTime: event.target.endTime.value,
+                primaryContact: event.target.primaryContact.value,
+                notes: event.target.notes.value,
+                receiveingAgency: {
+                    uid: raUid,
+                    confirmed: false
+                },
+                delivererGroup: {
+                    uid: dgUid,
+                    confirmed: false
+                },
+                requestTimestamp: Date.now()
+            };
+            let date = new Date(deliveryRequest.startDate);
+            let day = date.getDay();
+            let days = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
```

Comment:
i believe `Date.getDay()` returns 0-6 where 0 is Sunday, 1 is Monday

Reply:
That's what I thought too but I tested it out and 0 is Monday